### DISCUSSION TITLE
Difference between firefox extension and chrome extension behavior

### DISCUSSION THREAD

**Discussion by pinpontitit** [floccusaddon/floccus#1179] [Q&A] [answered]
I noticed that Firefox and Chrome have different behavior concerning extensions (at least the floccus one):

In firefox, when you click on floccus, 
a popup appear with the main screen, 
and if you click for example on "Secure your credential",
the main screen in the popup is replaced by the "Secure your Credential" screen.

In Chrome, when you click on floccus,
a popup appear with the main screen,
but if you click on "secure your credential"
it always open a new tab.
And it keeps opening new tab like this for every click on every button: "Secure your credentials" or "Settings" or "Floccus v.4.14", etc

I prefer the Firefox way of behaving. I think its more beautiful, it integrates well with the browser. It doesnt open unneccessary tabs, etc.
(There's only one situation where all-popup in Firefox is bad. Its when accessing external links.
For example when you click on "Funding" button and then any of the 4 options (Paypal, Github, Liberapay, patreon),
istead of opening the websites in tabs, it opens it in the popup which is impossible to work on. 
Could it be fixed by some code saying:
"When clicking in a button opening external links, always open in tab instead of staying in the popup "
?)

However the big problem with the firefox behavior is that it doesnt save the things you type.

If in firefox, i click on Secure your credentials and then start typing a passphrase, if I open another window, another app, or click somewhere else in the firefox browser, the popup is gone, and the passphrase I started entering is gone with it.

For example, imagine you want to go to your password manager to generate some random password, and come back to paste it. You have to go, once again, through all the steps.

Another example, that is really annoying:
When I am debugging floccus in Firefox Dev. gulp watch is running etc.
If I want to try some UI changes in some floccus screen I always have to come back to it from scratch.
For example I go to "Secure your credentials", I think the button should disappear, so I go to my code editor to try to delete some snippets, the go back to Firefox Dev, and puff! the popup is close and I have to open it again.

In Chrome it's easy: I can go back and forth from code editor to Chrome and the tab is always open and show in real time the changes.

So i decided to open this topic to express my hate of this LOL
Do you think there's a way of enabling auto-save in Firefox popup ?
Do you think there's a way of enabling tabs in Firefox instead of all-popup ?

Thanks

**Comment by marcelklehr** [comment, marked as answer]
This is curious. For me everything opens in new tabs in firefox and in chrome, which is how it's intended. Do you have specific settings about opening links in new tabs in firefox?

**Reply by pinpontitit** [reply]
In Firefox when I click on a link it opens in the current tab.
If I want to open a link in a new tab, I do right click then "open in a new tab"

However I think its not normal behavior for Firefox, and I remember I did tweak some values in "about:config".
But I did not remember if it was in my current Firefox or other Firefox-based browsers (Librewolf or Firefox < 100).

So I went in about:config, check the "browser.link.open_newwindow" value and all the other similar values. And surprisingly they're all set to default.
So I guess it was in another browser that I tweaked those. but still: when I click on a link it opens in and replace current tab.

Finally, I didn't know those options applied as well for the extensions links.

**Reply by marcelklehr** [reply]
> In Firefox when I click on a link it opens in the current tab.

That's normal though. It's the same for me. Some links have a link target specified which can mandate that the link be opened in a new window/tab, which is what floccus does. It's curious that your firefox doesn't seem to follow that instruction.

**Reply by pinpontitit** [reply]
> That's normal though. It's the same for me.

Ah ok, I wasnt sure.

Yes its strange that its behaving differently.

I use Firefox 100, with no special config.
